env/src/main/java/com/sap/cloud/security/config/k8s/K8sServiceConfigurationResolver.java

```diff
@@ -0,0 +1,154 @@
+/**
+ * SPDX-FileCopyrightText: 2018-2021 SAP SE or an SAP affiliate company and Cloud Security Client Java contributors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package com.sap.cloud.security.config.k8s;
+
+import com.sap.cloud.security.config.OAuth2ServiceConfiguration;
+import com.sap.cloud.security.config.OAuth2ServiceConfigurationBuilder;
+import com.sap.cloud.security.config.Service;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static com.sap.cloud.security.config.k8s.K8sConstants.*;
+
+/**
+ * The K8sServiceConfigurationResolver manages the service secret access from
+ * the file system.
+ */
+class K8sServiceConfigurationResolver {
+	private static final Logger LOGGER = LoggerFactory.getLogger(K8sServiceConfigurationResolver.class);
+
+	private String xsuaaPath;
+	private String iasPath;
+	private String serviceManagerPath;
+
+	K8sServiceConfigurationResolver() {
+		resolveServiceConfigurationPaths();
+	}
+
+	private void resolveServiceConfigurationPaths() {
+		this.xsuaaPath = getUserDefinedConfigurationPath(XSUAA_CONFIG_PATH);
+		this.iasPath = getUserDefinedConfigurationPath(IAS_CONFIG_PATH);
+		this.serviceManagerPath = getUserDefinedConfigurationPath(SM_CONFIG_PATH);
+		if (this.xsuaaPath == null) {
+			this.xsuaaPath = XSUAA_CONFIG_PATH_DEFAULT;
+		}
+		if (this.iasPath == null) {
+			this.iasPath = IAS_CONFIG_PATH_DEFAULT;
+		}
+		if (this.serviceManagerPath == null) {
+			this.serviceManagerPath = SERVICE_MANAGER_CONFIG_PATH_DEFAULT;
+		}
+	}
+
+	@Nullable
+	private String getUserDefinedConfigurationPath(String xsuaaConfigPath) {
+		// Resolves also empty string as null
+		if (System.getenv(xsuaaConfigPath) == null || System.getenv(xsuaaConfigPath).isEmpty()) {
+			return null;
+		}
+		return System.getenv(xsuaaConfigPath);
+	}
+
+	/**
+	 * Loads Service manager OAuth2 service configuration.
```

Comment:
as said this return type is imho strange => i would at least change the wording "OAuth2 service configuration" in the javadoc